### ISSUE TITLE
Global styles: Add color-contrast checks

### ISSUE THREAD

**Issue by aristath** [WordPress/gutenberg#28782]
Came up during a contributors day, from users testing FSE for the 1st time.

When editing a post, selecting white background with white text for a paragraph brings up a notice about the color-a11y.
We should have the same behavior while in global-styles, both for the per-block-type options and for the global options.

cc @moorscode

**Comment by aristath** [comment]
Closing this one in favor of #29457

**Comment by nosolosw** [comment]
There are two things here:

- One is the contrast checker for the block sidebar => the issue seems to be that the editor canvas is iframed, but the contrast checker feature is expected to work normally. I think we can leave https://github.com/WordPress/gutenberg/issues/29457 for this, as there's move momentum there.

- The other is the global sidebar => this is different in that we are not targeting a specific block, so we don't have blocks to check the contrast on but we have styles that affect the whole site. We'd need to check the whole document in this case. The advantage here is that we have access to structure data for colors, so perhaps we can do _some_ things. Unsure this is worth it, though.
